Curated groups and plots improvements - Part 1

labkey-bpatel opened the pull request:
#### Rationale
Improve organization of Group categories (Saved Groups and Curated Groups) by adding them to the Learn page and add link to the Learn > Groups grid in various appropriate places in the application for accessibility and readability.

Spec [here](https://docs.google.com/document/d/1EEik5Li2vCdLdEts3lBxwdS4XRwjDrJ4BcMTyejJJfk/edit).

Below changes are implemented in this PR.

#### Changes
* Create a query to display Groups data on the Learn grid.
* Add Groups to Learn grid as two grids using Ext's 'grouping' feature (one for 'My Saved Groups' and one for 'Curated groups') sharing the Columns.
* Add Name & Description, Studies, Species, Products and Assays.
* Enable Filtering and Search.
* Styling updates.
* Navigate user to GroupSummary page when user clicks a row in Learn > Groups grid (same as when user goes to group Details page from the home page by clicking on the group link).
* From Home page > Groups > Details or from Learn > Groups > Details, navigate user back to Learn > Groups grid for both the scenarios.

labkey-bpatel committed:
Create Groups query. Add Groups to Learn tab (col details wip)

labkey-bpatel committed:
Add collapsible 'Curated groups' and 'My saved groups' sections (styling and addition of other columns pending)

labkey-bpatel committed:
Display Studies label

labkey-bpatel committed:
Display Species

labkey-bpatel committed:
Display products and assays

labkey-bpatel committed:
Add description

labkey-bpatel committed:
Styling updates for collapsible button

labkey-bpatel committed:
Updates for column Filter and Search

labkey-bpatel committed:
Add groupTpl to the grouping feature to avoid regression in other places

labkey-bpatel committed:
Add groupTpl and customize collapsible group headers

labkey-bpatel committed:
Add placeholder for the Group Overview page

labkey-bpatel committed:
Handle Group navigation, remove newly added overview view, its not needed since using GroupSummary view now.

labkey-klum reviewed (approved):
Nice work leveraging the grid panel grouping feature!

There seem to be a lot of test failures in this branch as opposed to develop, I didn't investigate whether they were related to your changes but probably worth a look.

labkey-bpatel committed:
Bug fixes:
- Fix sorting on Studies, Species, Products, and Assays cols
- Fix positioning of expand/collapse button upon resizing
- Remove svg file with duplicate down arrow image

labkey-bpatel committed:
Update Learn header margin to fit all tabs.
